concept/type_/type_manager.rs

```diff
@@ -1182,136 +1577,625 @@ impl<Snapshot: WritableSnapshot> TypeManager<Snapshot> {
-        TypeWriter::storage_put_type_edge_property::<AnnotationUnique>(snapshot, edge, None)
+    pub(crate) fn unset_relates_overridden(
+        &self,
+        snapshot: &mut Snapshot,
+        relates: Relates<'static>,
+    ) -> Result<(), ConceptWriteError> {
+        TypeWriter::storage_delete_supertype(snapshot, relates.role().clone());
+        self.set_role_type_root_supertype(snapshot, relates.role());
+        TypeWriter::storage_delete_type_edge_overridden(snapshot, relates);
+        Ok(())
     }
 
-    pub(crate) fn delete_edge_annotation_unique<'b>(&self, snapshot: &mut Snapshot, edge: impl TypeEdgeEncoding<'b>) {
-        // TODO: Validation
-        TypeWriter::storage_delete_type_edge_property::<AnnotationUnique>(snapshot, edge)
+    pub(crate) fn set_relates_annotation_distinct(
+        &self,
+        snapshot: &mut Snapshot,
+        edge: Relates<'static>,
+    ) -> Result<(), ConceptWriteError> {
+        OperationTimeValidation::validate_relates_distinct_annotation_ordering(
+            snapshot,
+            edge.clone(),
+            None,
+            Some(true),
+        )
+        .map_err(|source| ConceptWriteError::SchemaValidation { source })?;
+
+        TypeWriter::storage_put_type_edge_property::<AnnotationDistinct>(snapshot, edge, None);
+        Ok(())
     }
 
-    pub(crate) fn set_edge_annotation_key<'b>(&self, snapshot: &mut Snapshot, edge: impl TypeEdgeEncoding<'b>) {
-        TypeWriter::storage_put_type_edge_property::<AnnotationKey>(snapshot, edge, None)
+    pub(crate) fn unset_edge_annotation_distinct<EDGE>(
+        &self,
+        snapshot: &mut Snapshot,
+        edge: EDGE,
+    ) -> Result<(), ConceptWriteError>
+    where
+        EDGE: TypeEdgeEncoding<'static> + InterfaceImplementation<'static> + Clone,
+    {
+        OperationTimeValidation::validate_unset_edge_annotation_is_not_inherited(
+            snapshot,
+            edge.clone(),
+            AnnotationCategory::Distinct,
+        )
+        .map_err(|source| ConceptWriteError::SchemaValidation { source })?;
+
+        TypeWriter::storage_delete_type_edge_property::<AnnotationDistinct>(snapshot, edge);
+        Ok(())
     }
 
-    pub(crate) fn delete_edge_annotation_key<'b>(&self, snapshot: &mut Snapshot, edge: impl TypeEdgeEncoding<'b>) {
-        TypeWriter::storage_delete_type_edge_property::<AnnotationKey>(snapshot, edge)
+    pub(crate) fn set_owns_annotation_unique(
+        &self,
+        snapshot: &mut Snapshot,
+        owns: Owns<'static>,
+    ) -> Result<(), ConceptWriteError> {
+        OperationTimeValidation::validate_owns_value_type_compatible_to_unique_annotation(
+            snapshot,
+            owns.clone(),
+            TypeReader::get_value_type_without_source(snapshot, owns.attribute().clone())?,
+        )
+        .map_err(|source| ConceptWriteError::SchemaValidation { source })?;
+
+        OperationTimeValidation::validate_set_edge_annotation_is_compatible_with_declared_annotations(
```

Comment:
I call this and the next functions for all `set_annotation`, so we might want to move to a "common" check. Like `Self::set_type_annotation()` and `Self::set_edge_annotation()` with these two functions + `TypeWrite::storage_put***`. 
Decided not to overdo refactoring for now

Reply:
Packed everything in common functions, looks much better.